src/main/java/org/fcrepo/auth/webac/WebACRolesProvider.java

```diff
@@ -193,6 +198,64 @@ public void deleteRoles(final Node node) throws RepositoryException {
     };
 
     /**
+     *  This maps a Collection of acl:agentClass values to a List of agents.
+     *  Any out-of-domain URIs are silently ignored.
+     */
+    private List<String> dereferenceAgentClasses(final Collection<String> agentClasses) {
+        final Session internalSession = sessionFactory.getInternalSession();
+        final IdentifierConverter<Resource, FedoraResource> translator =
+                new DefaultIdentifierTranslator(internalSession);
+
+        final List<String> members = new ArrayList<>();
+        agentClasses.stream()
+                    .distinct()
+                    .forEach(x -> {
+                        if (x.startsWith(FEDORA_INTERNAL_PREFIX)) {
+                            final FedoraResource resource = nodeService.find(
+                                internalSession, x.substring(FEDORA_INTERNAL_PREFIX.length()));
+                            members.addAll(getFoafMembers(translator, resource));
+                        } else if (x.startsWith(FOAF_NAMESPACE_VALUE)) {
```

Comment:
Thanks, I see where the `else if (x.startsWith(FOAF_NAMESPACE_VALUE))` comes from now; however, is this the logic we want? namely, to honor any term that happens to be in the `foaf` namespace as an `acl:agentClass`? 
If we are just expecting `foaf:Agent`, we should probably restrict the logic to that. 
If we are expecting any `rdf:type` that may define a category of agent, we should open the door wider.
But allowing any `foaf` terms seems like an odd middle-ground.